src/services/index.ts

```diff
@@ -1,51 +1,36 @@
 import { Api, Response as ApiResponse } from '../typings/Api';
 import { Handler } from '../typings/Handler';
-
+ 
 /** Begins the chain of API calls */
-export const begin = async (
-  apis: Api[],
-  onError: Handler,
-  interval?: number,
-  onResponse?: Handler
-) => {
-  const currentInterval = interval ? interval : 30000;
-
-  apis.forEach(async a => {
-    await makeCall(a, onError, onResponse || undefined);
-  });
-
-  /** Set the timeout and do it again */
-  setTimeout(
-    async () => await begin(apis, onError, currentInterval, onResponse || undefined),
-    currentInterval
-  );
+export const begin = async (apis: Api[], onResponse: Handler, interval?: number) => {
+ const currentInterval = interval ? interval : 30000;
+ 
+ apis.forEach(async a => {
+   await makeCall(a, onResponse);
+ });
+ 
+ /** Set the timeout and do it again */
+ setTimeout(async () => await begin(apis, onResponse, currentInterval), currentInterval);
 };
-
+ 
 /** Make the actual fetch and return a Response object */
-const makeCall = async (api: Api, onError: Handler, onResponse?: Handler) => {
-  await fetch(api.endpoint)
-    .then(apiResponse => handleResponse(api, apiResponse, onError, onResponse || undefined))
-    // tslint:disable-next-line
-    .catch(error => console.log('unhandled', error));
+const makeCall = async (api: Api, onResponse: Handler) => {
+ await fetch(api.endpoint)
+   .then(apiResponse => handleResponse(api, apiResponse, onResponse))
+   // tslint:disable-next-line
+   .catch(error => handleResponse(api, error, onResponse));
 };
-
+ 
 /**
- * Handles the Response
- */
-const handleResponse = (api: Api, response: Response, onError: Handler, onResponse?: Handler) => {
-  const apiResponse: ApiResponse = {
-    code: response.status,
-    message: response.statusText,
-  };
-
-  if (!response.ok) {
-    onError(api, apiResponse);
-    return;
-  }
-
-  if (onResponse) {
-    onResponse(api, apiResponse);
-  }
+* Handles the Response
+*/
+const handleResponse = (api: Api, response: Response, onResponse: Handler) => {
+ const apiResponse: ApiResponse = {
+   code: response.status,
+   message: response.statusText,
+ };
+ 
+ onResponse(api, apiResponse);
 };
-
-export default { begin, makeCall, handleResponse };
+ 
+export default { begin, makeCall, handleResponse };
```

Comment:
Please revert

Reply:
I fully reworked it according to your main idea